Static message analyzer fails to observe unexported constant in other file in same package

As a minimal test case, we can use `main.go`:

```go
package main

import (
	"log/slog"
)

const thisFileConstMsg = "Hello from main.go"

const localCopy = otherFileConstMsg

func main() {
	slog.Info(thisFileConstMsg)
	slog.Info(otherFileConstMsg)
	slog.Info(localCopy)
}
```

and `message.go`, in the same directory as `main.go`:
```go
package main

const otherFileConstMsg = "Hello from message.go"
```

If I `go run` the directory, I get the expected three log lines:

```
$ go run .
2025/07/09 14:28:15 INFO Hello from main.go
2025/07/09 14:28:15 INFO Hello from message.go
2025/07/09 14:28:15 INFO Hello from message.go
```

But if I try to analyze the file:

```shell
$ sloglint -static-msg=true ./_test/
.../main.go:13:12: message should be a string literal or a constant
exit status 3
```

If I comment or remove the `slog.Info(otherFileConstMsg)`, analysis succeeds.

From a quick inspection of sloglint source, I see that in `isStaticMsg`:

```
	case *ast.Ident: // e.g. const msg = "msg"; slog.Info(msg)
		return msg.Obj != nil && msg.Obj.Kind == ast.Con
```

we observe `msg.Obj.Kind` to be nil when the analysis fails.

I've observed this on the current main commit (0a750b14f11c89496b34f2c0f636f76e896b520e) and in tag v0.11.0 which is bundled in the current release of golang-ci.

--- a/sloglint.go
+++ b/sloglint.go
@@ -252,7 +252,7 @@ func visit(pass *analysis.Pass, opts *Options, node ast.Node, stack []ast.Node) 
 	msgPos := funcInfo.argsPos - 1
 
 	// NOTE: "With" functions have no message argument and must be skipped.
-	if opts.StaticMsg && msgPos >= 0 && !isStaticMsg(call.Args[msgPos]) {
+	if opts.StaticMsg && msgPos >= 0 && !isStaticMsg(pass.TypesInfo, call.Args[msgPos]) {
 		pass.Reportf(call.Args[msgPos].Pos(), "message should be a string literal or a constant")
 	}
 
@@ -440,17 +440,18 @@ func isContextInScope(info *types.Info, stack []ast.Node) bool {
 	return false
 }
 
-func isStaticMsg(msg ast.Expr) bool {
+func isStaticMsg(info *types.Info, msg ast.Expr) bool {
 	switch msg := msg.(type) {
 	case *ast.BasicLit: // e.g. slog.Info("msg")
 		return msg.Kind == token.STRING
 	case *ast.Ident: // e.g. const msg = "msg"; slog.Info(msg)
-		return msg.Obj != nil && msg.Obj.Kind == ast.Con
+		_, isConst := info.ObjectOf(msg).(*types.Const)
+		return isConst
 	case *ast.BinaryExpr: // e.g. slog.Info("x" + "y")
 		if msg.Op != token.ADD {
 			panic("unreachable") // only + can be applied to strings.
 		}
-		return isStaticMsg(msg.X) && isStaticMsg(msg.Y)
+		return isStaticMsg(info, msg.X) && isStaticMsg(info, msg.Y)
 	default:
 		return false
 	}

--- a/testdata/src/static_msg/another_file.go
+++ b/testdata/src/static_msg/another_file.go
@@ -1,0 +1,3 @@
+package static_msg
+
+const anotherConstMsg = "msg"

--- a/testdata/src/static_msg/static_msg.go
+++ b/testdata/src/static_msg/static_msg.go
@@ -41,3 +41,8 @@ func tests() {
 	slog.Info("msg" + fmt.Sprintf("msg"))                     // want `message should be a string literal or a constant`
 	slog.Info("msg" + constMsg + varMsg + fmt.Sprintf("msg")) // want `message should be a string literal or a constant`
 }
+
+func issue92() {
+	slog.Info(constMsg)
+	slog.Info(anotherConstMsg)
+}

